sympy/matrices/sparse.py

```diff
@@ -61,32 +135,78 @@ def __new__(cls, *args, **kwargs):
-                for key in args[2].keys():
-                    v = args[2][key]
+                def update(i, j, v):
+                    # update self._smat and make sure there are
+                    # no collisions
                     if v:
-                        self._smat[key] = self._sympify(v)
+                        if (i, j) in self._smat and v != self._smat[i, j]:
+                            raise ValueError('collision at %s' % ((i, j),))
+                        self._smat[i, j] = v
+                # manual copy, copy.deepcopy() doesn't work
+                for key, v in args[2].items():
+                    r, c = key
+                    if isinstance(v, SparseMatrix):
+                        for (i, j), vij in v._smat.items():
+                            update(r + i, c + j, vij)
+                    else:
+                        if isinstance(v, (Matrix, list, tuple)):
+                            v = SparseMatrix(v)
+                            for i, j in v._smat:
+                                update(r + i, c + j, v[i, j])
+                        else:
+                            v = self._sympify(v)
+                            update(r, c, self._sympify(v))
             elif is_sequence(args[2]):
-                if len(args[2]) != self.rows*self.cols:
-                    raise ValueError(
-                        'List length (%s) != rows*columns (%s)' %
-                        (len(args[2]), self.rows*self.cols))
-                flat_list = args[2]
+                flat = not any(is_sequence(i) for i in args[2])
+                if not flat:
+                    s = SparseMatrix(args[2])
+                    self._smat = s._smat
+                else:
+                    if len(args[2]) != self.rows*self.cols:
+                        raise ValueError(
+                            'Flat list length (%s) != rows*columns (%s)' %
+                            (len(args[2]), self.rows*self.cols))
+                    flat_list = args[2]
+                    for i in range(self.rows):
+                        for j in range(self.cols):
+                            value = self._sympify(flat_list[i*self.cols + j])
+                            if value:
+                                self._smat[i, j] = value
+            if self.rows is None:  # autosizing
+                k = self._smat.keys()
+                self.rows = max([i[0] for i in k]) + 1 if k else 0
+                self.cols = max([i[1] for i in k]) + 1 if k else 0
+            else:
+                for i, j in self._smat.keys():
+                    if i and i >= self.rows or j and j >= self.cols:
+                        r, c = self.shape
+                        raise ValueError(filldedent('''
+                            The location %s is out of designated
+                            range: %s''' % ((i, j), (r - 1, c - 1))))
+        else:
+            if (len(args) == 1 and isinstance(args[0], (list, tuple))):
```

Comment:
Skip outer brackets?

Reply:
What I intend to do is hand the case where args is `([...],)` then `v = [...]`

Reply:
I was more thinking:
`if len(args) == 1 and isinstance(args[0], (list, tuple)):`
which should be the same.